rfd/0134-cloud-first-deployment.md

```diff
@@ -0,0 +1,203 @@
+                                |
+                                |
+            14.0.0-cloud.1      14.0.0-cloud.10    15.0.0-cloud.0     15.0.0-cloud.1     16.0.0-cloud.0
+master -----|-------....--------|------------------|-------....-------|------------------|--------------....---->
+                                                                      |
+                                                                      |
+                                                           branch/v15 |---------|---------|---------...---->
+                                                                      15.0.0    15.0.1    15.0.2
+```
+
+Once a self-hosted release branch is cut, subsequent self-hosted minor/patch
+releases are published without any changes to the current process, but on a
+less frequent cadence than Cloud releases off of master.
+
+### Cloud release artifacts
+
+The self-hosted release and promotion process is slow and takes 2-3 hours
+because it builds a full suite of artifacts. Cloud users need most of these
+artifacts as well for the agents they run on their own infrastructure, however
+Cloud control plane needs a very limited subset of all artifacts. For this
+reason, the Cloud release and promotion process will be split in 2 parts:
+
+- The workflow that builds artifacts required for the control plane: amd64
+  binaries and Docker images.
+- The workflow that builds remaining artifacts such as deb/rpm packages and
+  Helm charts.
+
+This will allow us to produce rapid releases that affect only control plane
+services (auth, proxy) while still having the ability to push remaining artifacts
+when needed.
+
+Both workflows will run in Github Actions and will reuse existing bits of the
+full self-hosted release pipeline.
+
+### Cloud promotion process
+
+The Cloud release promotion process will be staged, with a different class of
+tenants being upgraded at each stage:
+
+1. Canary
+2. Internal
+3. Trials
+4. Team & Enterprise
+
+The canary tenants will be a set of precreated clusters with some data in them
+which engineers will use for connecting their agents and testing their features.
+Canary clusters will be updated automatically once a Cloud release is cut.
+
+Before promoting the release further, engineers will test any significant change
+in the cloud staging environment and then explicitly sign off that the change is
+ready for deployment by applying a label (e.g. `cloud/verified`) on the master
+PR corresponding to the change. The release manager then will approve the
+further rollout.
+
+Once the change is rolled out to production tenants, engineers will be expected
+to monitor error rates and Cloud metrics relevant to their changes.
+
+Let's now look at some of the edge case scenarios in the release process and
+how we'll be handling them.
+
+### Behavior-incompatible changes
+
+Teleport's component compatibility guarantees become somewhat obsolete in a
```

Comment:
Yes and no. If we're still cutting self-hosted releases, then people are going to expect some level of backwards compatibility between "major releases" of the self-hosted product. 

We will need to define and manage our backwards compatibility guarantees in a way that works for both the cloud-first and self-hosted release cadences (even if the self-hosted compatibility guarantee turns out to be "well, good luck with that...".)

Reply:
Right, this is referring to Cloud compatibility specifically. For self-hosted releases I don't expect we can change our compatibility guarantees, added a sentence to clarify this below.